### DISCUSSION TITLE
Configuring IP ranges

### DISCUSSION THREAD

**Discussion by dercorn** [mysticaltech/terraform-hcloud-kube-hetzner#858] [Q&A] [answered]
Hello,

I want to set specific IP ranges for the cluster VMs.

It looks like network_ipv4_cidr and cluster_ipv4_cidr are intended for this purpose. However, the comments above these parameters state that it is "highly advised against". Could you please elaborate why that is so?

Also, I tried applying new IP ranges and that did not seem to work.

Best regards

**Comment by mysticaltech** [comment]
@dercorn You have to get it right. From memory, cluster_ipv4_cidr needs to be a subset of network_ipv4_cidr. If that does not work, have a look over at locals.tf how the subnets are calculated, you will understand better how to safely change these IP ranges.

**Comment by dercorn** [comment]
Thank you for your reply!
I left the network_ipv4_cidr at its "10.0.0.0/8" default.
With cluster_ipv4_cidr = "10.0.5.0/24", I would expect IPs in the range from 10.0.5.1 to 10.0.5.254.

However, when I deploy a cluster with these settings, I get IPs like:
10.1.0.101
10.2.0.101
10.253.0.101
10.254.0.101
10.255.0.101

Looking at locals.tf in line 90, I see that 10.1.0.0/16 is the default for control plane nodes.
This results in IP addresses ranging from 10.1.0.1 to 10.1.255.254.

Then, looking at line 92 which looks like this:
network_ipv4_subnets = [for index in range(256) : cidrsubnet(var.network_ipv4_cidr, 8, index)]

This creates a list of 256 subnets ranging from 10.0.0.0/16 to 10.255.0.0/16.

Long story short, if I want to have all the cloud VMs within a 10.0.5.0/24 subnet, it will not be possible without a code change. Or am I missing something?

**Reply by mysticaltech** [reply, marked as answer]
Exactly, currently it would not be possible without code change.

**Comment by dercorn** [comment]
Thank you! That answers my question.
